core/src/main/java/org/elasticsearch/common/ParseField.java

```diff
@@ -33,12 +33,24 @@
     private static final DeprecationLogger DEPRECATION_LOGGER = new DeprecationLogger(Loggers.getLogger(ParseField.class));
 
     private final String name;
+    private final String[] alternativeNames;
     private final String[] deprecatedNames;
```

Comment:
Can we add a bit of docs or code comments... somewhere... anywhere... to this file, to explain the distinction of alternative vs deprecated.

It makes total sense given what you have explained on the PR here, but just looking at the code, without the context of the PR, it can be a bit confusing. 

Matching logic and so on in this class is already confusing (given it has allReplacedWith and allNames, neither of which i have a clue what is doing). So it could really help to disambiguate it!